src/app/core/components/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -20,7 +20,7 @@ export class BreadcrumbsComponent {
         while (currentRoute?.firstChild) {
           const { routeConfig, data, url } = currentRoute.firstChild;
 
-          if (data.hasOwnProperty('title') && routeConfig?.path) {
+          if (data.hasOwnProperty('title') && routeConfig?.path && !breadcrumbs.map(b => b.label).includes(data['title'])) {
```

Comment:
You can check this with a single method to reduce complexity from `On*2` to `On`.
```suggestion
          if (data.hasOwnProperty('title') && routeConfig?.path && !breadcrumbs.some(b => b.label === data['title'])) {
```

Reply:
Nice